### ISSUE TITLE
DateFieldInput: The component does not inform the user when the date is invalid.

### ISSUE THREAD

**Issue by olehnoskov-cengage** [cengage/react-magma#2407]
DatePicker with isDateFieldInput has some range:
**MIN_YEAR = 1900**
**MAX_YEAR = 2099**
When the user types a year that is out of this range, the date is considered invalid.

However, we do not inform users about this.
For now, in this case, we handle when the user changes the year value by using ArrowUp or ArrowDown. We do not restrict the user type to any year values.

Fix `for` issue.

<img width="1779" height="1029" alt="Image" src="https://github.com/user-attachments/assets/3f35f4bb-f600-4a48-9f92-9d1d0e565d3d" />

[Details](https://cengage.slack.com/archives/C0A9TSB75P0/p1774978755253869)

**Comment by nikitaorliak-cengage** [comment]
**Tested on:**
- V5: https://storybook-preview-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/datepicker--default
- V4: https://storybook-preview-v4-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/datepicker--default

Everything works as expected. No bugs found.
